### PR TITLE
Update field mutation error message

### PR DESCRIPTION
**Pull request by PgBiel**
First follow-up for #790.
Since the idea is to make the new 'static fields' for types mutable in the future, I thought of updating the current (no longer very helpful) field mutation error, which just says "expected dictionary, found [type]". I changed the error to:

1. If the type has its own field getters, and thus no 'static fields' (symbols, functions and modules): "cannot mutate fields on [type]"  (aligned with the field access error "cannot access fields on [type]")
2. If the type does not have any field getters or 'static fields' (such as numbers, 'none', ...): "[type] does not have accessible fields"
3. If the type does have 'static fields': "fields on [type] are not yet mutable" (since the plan is to make all of them mutable anyway), followed by a hint "try creating a new [type] with the updated field value instead".

This is mostly done to guard against possible user confusion, since actual field mutation might not be ready before the next release.

Feel free to share opinions, suggest changes etc. I added a few tests for the PR (one of them replaces an older test from `dict.typ`). The idea is that the last test, for stroke mutation, will eventually be removed (as it should eventually work).

### CODE REVIEW

File: tests/typ/compiler/field.typ [hunk 1]
```diff
@@ -148,3 +148,18 @@
 #test((top + start).y, top)
 #test((bottom + end).y, bottom)
 #test((horizon + center).y, horizon)
+
+---
+#{
+  let object = none
+  // Error: 3-9 cannot mutate fields on none
```
Comment (laurmaedje) [inline tests/typ/compiler/field.typ:155]: is this really the clearest error message? isn't it even one step more. not just that you cannot mutate fields, none doesn't even _have_ fields.

Comment (PgBiel) [reply tests/typ/compiler/field.typ:155]: Yeah, I agree the error message could be more descriptive. Initially, in this PR, I had only distinguished between types with static fields and types without static fields, as it's simple to do that (a simple call to `fields::fields_on`). I can probably make this more fine-grained, however, using a proper `match` in the error message generation.

Will update the code later. Thanks for the feedback!

Comment (PgBiel) [reply tests/typ/compiler/field.typ]: Ok, should be better now :+1:

**Comment by PgBiel** [comment]
@laurmaedje this should be OK to merge now; feel free to add further suggestions if needed. :+1:

**Comment by laurmaedje** [comment]
Thanks for the reminder (and the PR)!

**Pull request merged**